hudi-common/src/main/java/org/apache/hudi/metadata/HoodieTableMetadataUtil.java

```diff
@@ -1046,40 +1048,14 @@ private static <T> ClosableIterator<BufferedRecord<T>> getLogRecords(List<String
           .withMetaClient(datasetMetaClient)
           .withAllowInflightInstants(true)
           .withRecordBuffer(recordBuffer)
-          .build();
-      return new CloseableLogRecordsIterator<>(mergedLogRecordReader, recordBuffer);
+          .build()) {
+        // initializes the record buffer with the log records
+        return recordBuffer.getLogRecordIterator();
```

Comment:
This is an existing issue, which we should track separately.  `recordBuffer.getLogRecordIterator()` invokes `ExternalSpillableMap#values` which puts all in-memory and disk-spilled records to a in-memory list which can cause OOM.